src/Lykke.Service.Nem.Api/Services/NemApi.cs

```diff
@@ -0,0 +1,311 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Common;
+using io.nem1.sdk.Infrastructure.HttpRepositories;
+using io.nem1.sdk.Infrastructure.Imported.Client;
+using io.nem1.sdk.Model.Accounts;
+using io.nem1.sdk.Model.Mosaics;
+using io.nem1.sdk.Model.Transactions;
+using io.nem1.sdk.Model.Transactions.Messages;
+using Lykke.Service.BlockchainApi.Contract;
+using Lykke.Service.BlockchainApi.Contract.Common;
+using Lykke.Service.BlockchainApi.Sdk;
+using Lykke.Service.BlockchainApi.Sdk.Domain.DepositWallets;
+using Newtonsoft.Json;
+
+namespace Lykke.Service.Nem.Api.Services
+{
+    public class NemApi : IBlockchainApi
+    {
+        const char AddressSeparator = '$';
+        const string XEM = "nem:xem";
+
+        readonly static DateTime _nemesis = 
+            new DateTime(2015, 03, 29, 0, 6, 25, 0, DateTimeKind.Utc);
+
+        readonly string _nemUrl;
+        readonly string _explorerUrl;
+        readonly int _requiredConfirmations;
+        readonly int _expiresInMinutes;
+        readonly DepositWalletRepository _deposits;
+
+        public NemApi(string nemUrl, string explorerUrl, int requiredConfirmations, int expiresInMinutes, DepositWalletRepository deposits)
+        {
+            _nemUrl = nemUrl;
+            _explorerUrl = explorerUrl;
+            _requiredConfirmations = requiredConfirmations;
+            _expiresInMinutes = expiresInMinutes;
+            _deposits = deposits;
+        }
+
+        public Task<bool> AddressIsExistAsync(string address) => Task.FromResult(AddressIsValid(address));
+
+        public bool AddressIsValid(string address)
+        {
+            try
+            {
+                return Address.CreateFromEncoded(address.Split(AddressSeparator)[0]) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public async Task<string> BroadcastTransactionAsync(string signedTransaction)
+        {
+            var result = await new TransactionHttp(_nemUrl).Announce(SignedTransaction.FromJson(signedTransaction));
+
+            // see https://nemproject.github.io/#nemRequestResult 
+            // for meaning of the code 
+
+            switch (result.Code)
+            {
+                case 0:
+                case 1:
+                    // success
+                    break;
+                case 3:
+                case 9:
+                    throw new BlockchainException(BlockchainErrorCode.BuildingShouldBeRepeated, "Transaction expired");
+                case 7:
+                    throw new BlockchainException(BlockchainErrorCode.BuildingShouldBeRepeated, "Transaction hash duplicated");
```

Comment:
How this case is distinguished of broadcast retry? What if tx was successfully broadcasted to the node, response processing was failed and then the tx is broadcasted again due to broadcast retry?

Reply:
If we send the same transaction multiple times then the answer is `0`: 
> Neutral result. A typical example would be that a node validates an incoming transaction and realizes that it already knows about the transaction. In this case it is neither a success (meaning the node has a new transaction) nor a failure (because the transaction itself is valid).

Error code `7` is returned when transactions are different but their hashes are the same. New transaction is rejected in this case. Actually, I've been not able to reproduce an error with code `7`.